src/persist/src/indexed/mod.rs

```diff
@@ -202,89 +236,446 @@ impl<K: Data, V: Data, U: Buffer, L: Blob> Indexed<K, V, U, L> {
             }
         };
 
-        // TODO: actually physically delete the future and trace batches.
-        let future = self.futures.remove(&mapping.1);
-        let trace = self.traces.remove(&mapping.1);
-
-        // Sanity check that we actually removed the future and trace for this
-        // stream.
-        debug_assert!(future.is_some());
-        debug_assert!(trace.is_some());
-
-        self.graveyard.push(mapping);
+        self.ids.remove(id_str);
+        self.graveyard.insert(id_str.to_string(), id);
+        self.seal_frontiers.remove(&id);
+        self.since_frontiers.remove(&id);
+        self.listeners.remove(&id);
+        let entry: BufferEntry<K, V> = BufferEntry::Destroy(id, id_str.to_string());
+        let mut entry_bytes = Vec::new();
```

Comment:
in this and every other function on `IndexedClient` we need to be careful to do the process in three stages

1. determine if the operation is valid, if not return
2. determine if the operation can be sent downstream to `IndexedCore`, if not return
3. make local modifications on `IndexedClient` metadata to reflect the new state after the operation